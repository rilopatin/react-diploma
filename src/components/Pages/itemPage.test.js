import React from 'react'
import ItemPage from './itemPage'
import {shallow} from 'enzyme'
import ErrorMessage from '../errorMessage/errorMessage'

//describe('Testing <ItemPage/>', () => {
    const component = shallow(<ItemPage match={{params: {id: 1}, isExact: true, path: "", url: ""}}/> )

    describe('Testing <ItemPage/>', () => {
        it('ItemPage has been rendered correctly', () => {
            expect(component).toMatchSnapshot()
        });
    // describe('renders props <ItemPage/>', () => {
    //     console.log(component.instance().props)
    // })

        it('ItemPage state "loading" is true', () => {
            expect(component.state().loading).toBeTruthy()
        });
        it('ItemPage state "error" is false', () => {
            expect(component.state().error).toBeFalsy()
        });
    });
  //  })
    describe('Handlers tests', () => {
        it('testing onError', () => {
            component.instance().onError(<ErrorMessage code={'404'}/>);
            expect(component.state().loading).toBeFalsy();
            expect(component.state().error).toBeTruthy()
        });
        // it('testing updateItem', () => {
        //     component.instance().updateItem()
        //     expect(component.state().loading).toBeTruthy()
        // })
    })
