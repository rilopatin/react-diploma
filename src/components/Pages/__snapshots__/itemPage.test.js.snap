// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ItemPage/> ItemPage has been rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemPage
    match={
      Object {
        "isExact": true,
        "params": Object {
          "id": 1,
        },
        "path": "",
        "url": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <Spinner />,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <Spinner />,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
